### DISCUSSION TITLE
Listing Digital and Physical Products in Same Listing

### DISCUSSION THREAD

**Discussion by samborambo305** [etsy/open-api#1083] [Q&A]
Use Case: We offer print-on-demand services for artists. Many of these artists sell digital downloads. We would like allow artists to create variations of physical products and add them to existing listings of the digital downloads listings the artist is already selling. Ideally, the Etsy customer would visit a listing where they can choose the variation that suits their need by choosing "digital download" or one of the physical product options we offer. 

Background: 

Which of the following options are feasible?

Option 1: Convert digital listings to physical listings using the updateListing endpoint. From there, create variations with names like "Digital Download", "Unframed 12x16", "Unframed 24x36" for instance.

Issues with option 1:
- The documentation is unclear as to if you can convert a digital listing to a physical listing.
- On Shop Manager Etsy sellers can create new listings and choose either "physical item" or "digital files", there doesn't seem to be a way where a user can sell both digital and physical items unless you use option one.
- Assuming the conversion works (or if you do it manually using shop manager) You will have a "digital download" that is named digital download but is technically considered a physical item by Etsy. Thus, the automatic download isn't automatically sent. 

Option 2: There is a listing_type prop that allows for "physical", "digital", or "both". I'm am going to test whether or not this allows a listing to sell both digital and physical products, if so this is the preferred option. But I don't know if the API allows for all of these operations. 

Issues with option 2:
- The API may not support this or I am misunderstanding the documentation. 

Option 3: Convert existing digital downloads to physical variations

Issues with option 3:
- This would no longer allow the seller to sell digital downloads
- This could negatively impact the search rankings of their products

Option 4: Create duplicate listings for physical variations of digital products. 

Issues with Option 4:
- This would create 2x the number of listings
- This would be annoying for customers who may come across one listing (digital) but need to find the physical listing if they wish to purchase physical. 

Thoughts?

**Comment by samborambo305** [comment]
@etsyachristensen Do you have any idea on this?

**Comment by samborambo305** [comment]
@nateswart @pnaf

**Comment by samborambo305** [comment]
@jasonpincin Could you please take a look at this? Sorry for tagging you guys. This is just a huge blocker at the moment and I don't wanna get fired! haha

**Comment by emergingdzns** [comment]
@samborambo305 you should be able to use the `both` setting on the listing. But doing so would not make it so that it will only send the digital file for specific variants. The digital file(s) still get sent for all I believe. The `both` setting is (I believe) to make it so that the seller can automatically send a digital form followed by the physical form. But I think that sellers also use this a way to auto send a template file to the buyer to manipulate and send back to then have the physical product created (such as custom vinyl stickers or something). I don't see a reason it couldn't be used the way you are looking to do it, you just have to know that the digital file will get sent. If you don't want it to be sent, you could remove the files and manually send them after a digital variant purchase but that could be very time consuming depending on the order volume.

The most sure fire solution really is to do option 4 and create physical listings separate from the digital listings. You could then add links to the alternate version of each of the products in the description so that buyers and go from the digital to the physical listing and vice versa.

If you're doing this all via the API anyway it shouldn't be too much to have your system make to calls to the create endpoint and create two different versions. Not sure how hard it would be to manage on your backend though.

**Reply by ninjacell81** [reply]
Is there a way a user can choose "both" on Etsy's listing manager? I can only see the option to choose between digital and physical. Am I missing something?

**Comment by samborambo305** [comment]
Thanks, that is very helpful. I really appreciate it.

I think that leaves us here:
1.) Option 1 - We will try to create listing using the "both" setting. This means converting existing listings and also creating new ones. We think this is best because it would lower friction related to upselling. If this means that digital files will be sent regardless, then I don't think that's too much of a problem. We will test this internally and report back, perhaps this is an issue others will face in the future.

2) Option 4 - we will implement this as an alternative as well. 

Just to clarify, can you tell me if you're looking at option 1 the same way?
- If the listing is "both" then that would allow us to add multiple variants, something that is not possible using a digital listing. This will then allow the listing to have options like "digital", "framed small", "framed large" etc. Once a customer purchases any of the non-digital listings, the digital file will be sent regardless but we will process that order for physical fulfillment. Other than email dev support (which hasn't been answering us) do you know how else to get clarification on this issue? Perhaps there's someone we can pay to walk through this with?
